lib/remap-functions/src/to_syslog_severity.rs

```diff
@@ -0,0 +1,140 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToSyslogSeverity;
+
+impl Function for ToSyslogSeverity {
+    fn identifier(&self) -> &'static str {
+        "to_severity"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(ToSyslogSeverityFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToSyslogSeverityFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToSyslogSeverityFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let level_bytes = self.value.execute(state, object)?.try_bytes()?;
+
+        let level = String::from_utf8_lossy(&level_bytes);
+
+        // Severity levels: https://en.wikipedia.org/wiki/Syslog#Severity_level
+        let severity = match &level[..] {
+            "emerg" | "panic" => Ok(0),
+            "alert" => Ok(1),
+            "crit" => Ok(2),
+            "err" | "error" => Ok(3),
+            "warning" | "warn" => Ok(4),
+            "notice" => Ok(5),
+            "info" => Ok(6),
+            "debug" => Ok(7),
+            _ => Err(format!("level {} not valid", level)),
+        };
+
+        match severity {
+            Ok(severity) => Ok(Value::from(severity)),
+            Err(e) => Err(Error::Call(e)),
```

Comment:
Nite: Just noting that in the other PR (#5503) the `Error::Call` was unneeded due to using ` Err(Error::from(...))` above. I don't have strong opinions on which one is better, but it'd be nice to be consistent.

Reply:
Good catch. Will fix.